torch/_refs/nn/functional/__init__.py

```diff
@@ -87,6 +88,26 @@ def elu(
     return refs.where(refs.gt(a, 0), a, rhs)
 
 
+@register_decomposition(torch.ops.aten.leaky_relu)
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a",),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT
+)
+def leaky_relu(a: TensorLikeType, negative_slope: float = 0.01, inplace: bool = False) -> TensorLikeType:
+    """
+    Reference implementation of torch.nn.functional.hardswish
+    """
+
+    if inplace:
+        raise NotImplementedError
+
+    python_type = utils.dtype_to_type(a.dtype)
```

Comment:
I'm thinking of avoiding this by putting this check into the elementwise type promo decorator, but it might make the decorator confusing to have it doing so much

Reply:
yeah, that could be nice.